File: salao-backend/src/main/java/com/salao/modules/marca/MarcaService.java
package com.salao.modules.marca;

import com.salao.modules.produto.ProdutoRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class MarcaService {

    private final MarcaRepository repository;
    private final ProdutoRepository produtoRepository;

    public List<MarcaResponseDTO> listar() {
        return repository.findAll().stream().map(MarcaResponseDTO::from).toList();
    }

    public MarcaResponseDTO buscarPorId(Long id) {
        return MarcaResponseDTO.from(buscarEntidade(id));
    }

    public MarcaResponseDTO criar(MarcaRequestDTO dto) {
        var marca = Marca.builder()
                .marca(dto.marca())
                .ativo(dto.ativo() != null ? dto.ativo() : true)
                .build();
        return MarcaResponseDTO.from(repository.save(marca));
    }

    public MarcaResponseDTO atualizar(Long id, MarcaRequestDTO dto) {
        var marca = buscarEntidade(id);
        marca.setMarca(dto.marca());
        if (dto.ativo() != null) marca.setAtivo(dto.ativo());
        return MarcaResponseDTO.from(repository.save(marca));
    }

    public void deletar(Long id) {
        buscarEntidade(id);
        if (produtoRepository.existsByMarcaId(id))
            throw new RuntimeException("Não é possível excluir: existem produtos vinculados a esta marca");
        repository.deleteById(id);
    }

    private Marca buscarEntidade(Long id) {
        return repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Marca não encontrada"));
    }
}

File: salao-frontend/src/services/marcaService.ts
import axios from 'axios';

const API = 'http://localhost:8080/api';

export interface Marca {
  id: number;
  marca: string;
  ativo: boolean;
}

export interface MarcaRequest {
  marca: string;
  ativo: boolean;
}

export const marcaService = {
  listar:      ()                               => axios.get<Marca[]>(`${API}/marcas`),
  buscarPorId: (id: number)                     => axios.get<Marca>(`${API}/marcas/${id}`),
  criar:       (dto: MarcaRequest)              => axios.post<Marca>(`${API}/marcas`, dto),
  atualizar:   (id: number, dto: MarcaRequest)  => axios.put<Marca>(`${API}/marcas/${id}`, dto),
  deletar:     (id: number)                     => axios.delete(`${API}/marcas/${id}`),
};
